src/main/kotlin/com/sourcegraph/cody/context/ui/EnhancedContextPanel.kt

```diff
@@ -0,0 +1,128 @@
+                          agent.server.commandExecute(cmd).get()
+                        } catch (e: Exception) {
+                          val errMsg = e.message ?: e.toString()
+                          Messages.showErrorDialog(
+                              CodyBundle.getString("context-panel.error-message").fmt(errMsg),
+                              CodyBundle.getString("context-panel.error-title"))
+                        } finally {
+                          indicator.stop()
+                          isReindexingInProgress.set(false)
+                        }
+                      }
+                    })
+          }
+        }
+
+        override fun isEnabled(): Boolean {
+          return !isReindexingInProgress.get()
+        }
+
+        override fun isDumbAware(): Boolean {
+          return true
+        }
+
+        private val isReindexingInProgress = AtomicBoolean(false)
+      }
+
+  private val helpButton =
+      object :
+          ToolbarDecorator.ElementActionButton(
+              CodyBundle.getString("context-panel.help-button-name"), AllIcons.Actions.Help) {
+        override fun actionPerformed(p0: AnActionEvent) {
+          BrowserUtil.open("https://docs.sourcegraph.com/cody/core-concepts/keyword-search")
+        }
+
+        override fun isDumbAware(): Boolean {
+          return true
+        }
+      }
+
+  private val tree = run {
+    val treeRoot = CheckedTreeNode(CodyBundle.getString("context-panel.tree-root"))
+    val localProject = CheckedTreeNode(CodyBundle.getString("context-panel.tree-local-project"))
+    val currentRepo =
+        object : CheckedTreeNode(project) {
+          override fun isChecked(): Boolean {
+            return isEnhancedContextEnabled.get()
+          }
+
+          override fun setChecked(checked: Boolean) {
+            isEnhancedContextEnabled.set(checked)
+            CodyAgentService.getInstance(project).restartAgent(project)
+          }
+        }
+    localProject.add(currentRepo)
+    treeRoot.add(localProject)
+    CheckboxTree(ContextRepositoriesCheckboxRenderer(), treeRoot)
+  }
+
+  private val toolbarPanel = run {
+    val borderColor = ColorUtil.brighter(UIUtil.getPanelBackground(), 3)
+    val lightBorder = BorderFactory.createMatteBorder(1, 0, 0, 1, borderColor)
+    val titledBorder = TitledBorder(lightBorder, CodyBundle.getString("context-panel.panel-name"))
+
+    ToolbarDecorator.createDecorator(tree)
+        .disableUpDownActions()
+        .addExtraAction(refreshButton)
+        .addExtraAction(helpButton)
+        .setPreferredSize(Dimension(0, 20))
+        .setToolbarPosition(ActionToolbarPosition.LEFT)
+        .setPanelBorder(titledBorder)
+        .setScrollPaneBorder(BorderFactory.createEmptyBorder())
+        .setToolbarBorder(BorderFactory.createEmptyBorder())
+        .createPanel()
+  }
+
+  init {
+    layout = VerticalFlowLayout(VerticalFlowLayout.BOTTOM, 14, 0, true, false)
+    tree.expandRow(0)
+
+    add(toolbarPanel)
+  }
+}
```

Comment:
It's a bit hard to figure out what the execution order in this class. It looks like it was written backwards. :smile:

My proposals:
* Move init to top (as is should be). 
* Classes here don't share states so my suggestion is to replace anonymous classes with private static classes like "RefreshButton()", "HelpButton()". This inner classes should be places at bottom.

Reply:
What you propose is actually nt possible :)
Order of execution/creation is exactly as written.
Because of that `init` cannot be moved to the beginning as it would then operate on not-yey-existing-variables.
Classes I can move.

Reply:
Thanks for explanation. I still believe that replacing anonymous classes with private static inner classes will be a better solution. Ofc I leave the decision up to you.

Reply:
I moved them to the separate classes.
If we want to make them finer classes I'm fine with that as well, I can move it later.